docs/src/index.rst

```diff
@@ -1,10 +1,8 @@
-Welcome to metatensor-models's documentation!
+Welcome to metatensor-models' documentation!
```

Comment:
also, in the title, the `'s` was a typo, right? 
I am not good at English grammar